app/src/androidTest/java/com/github/HumanLearning2021/HumanLearningApp/view/SearchTest.kt

```diff
@@ -0,0 +1,213 @@
+
+
+    @get:Rule
+    val activityScenarioRule: ActivityScenarioRule<MainActivity> = ActivityScenarioRule(
+        Intent(
+            ApplicationProvider.getApplicationContext(),
+            MainActivity::class.java
+        )
+    )
+
+    @BindValue
+    @Demo2Database
+    val dbManagement: DatabaseManagement = DefaultDatabaseManagement(DummyDatabaseService())
+
+    private val dummyDatasets = runBlocking { dbManagement.getDatasets() }
+
+    @Before
+    fun setup() {
+        hiltRule.inject()
+        Intents.init()
+    }
+
+    @After
+    fun cleanUp() {
+        Intents.release()
+        activityScenarioRule.scenario.close()
+    }
+
+    @Test
+    fun datasetsOverviewInputEmptyStringYieldsAllResults() {
+        navigateToDatasetsOverview()
+        onView(withId(R.id.action_search)).perform(click(), typeText("    "))
+        onView(withId(R.id.DatasetList_list)).check(
+            ViewAssertions.matches(
+                ViewMatchers.hasChildCount(dummyDatasets.size)
+            )
+        )
+    }
+
+    @Test
+    fun datasetsOverviewSearchByKeyWordYieldsCorrectResult() {
+        navigateToDatasetsOverview()
+        onView(withId(R.id.action_search)).perform(click(), typeText("kitchen"))
+        onView(withId(R.id.DatasetList_list)).check(
+            ViewAssertions.matches(
+                ViewMatchers.hasChildCount(1)
+            )
+        )
+    }
+
+    @Test
+    fun datasetsOverviewSearchNotFoundYieldsNoResult() {
+        navigateToDatasetsOverview()
+        onView(withId(R.id.action_search)).perform(click(), typeText("hi"))
+        onView(withId(R.id.DatasetList_list)).check(
+            ViewAssertions.matches(
+                ViewMatchers.hasChildCount(0)
+            )
+        )
+    }
+
+    @Test
+    fun datasetsOverviewCanClickOnSubsetOfDatasetsMatchingSearch() {
+        navigateToDatasetsOverview()
+        onView(withId(R.id.action_search)).perform(click(), typeText("kitchen"))
+
+        onView(withId(R.id.DatasetList_list))
+            .perform(
+                RecyclerViewActions.actionOnItemAtPosition<DatasetListRecyclerViewAdapter.ListItemViewHolder>(
+                    0,
+                    click()
+                )
+            )
+        assertCurrentFragmentIsCorrect(R.id.displayDatasetFragment)
+    }
+
+    @Test
+    @Ignore // haven't found a way to clear the text
+    fun datasetsOverviewTypeTextAndThenClearYieldsAllResults() {
+        navigateToDatasetsOverview()
+        onView(withId(R.id.action_search)).perform(click(), typeText("kitchen"), clearText())
+        onView(withId(R.id.DatasetList_list)).check(
+            ViewAssertions.matches(
+                ViewMatchers.hasChildCount(dummyDatasets.size)
+            )
+        )
+    }
+
+    @Test
+    fun learningDatasetSelectionInputEmptyStringYieldsAllResults() {
+        navigateToLearningDatasetSelection()
+        onView(withId(R.id.action_search)).perform(click(), typeText("    "))
```

Comment:
"     " is not really an empty string, it's a string containing space characters. But I see what you mean. Maybe make it more explicit that spaces are treated as empty by the search

Reply:
sure, I wrote a separate test for this now.